### PR TITLE
Add eslint

### LINKED ISSUE #438: Lint JS (more aggressively) with ESLint
Note: if time permits, Matt will do this with the interns live!

### PR DESCRIPTION
**Pull request by reginawang99**
closes #438

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    // No multiple empty lines
+    'no-multiple-empty-lines': ['error'],
+
+    // Max line lengths
+    'max-len': ['error', {
+      code: 120,
+      ignoreUrls: true, // Most common reason to disable it
+      ignoreStrings: true, // These are not fantastic but necessary for error messages
+      ignoreTemplateLiterals: true,
+      ignoreComments: true,
```
Comment (lumisphere902) [inline .eslintrc.js:106]: I think we can enforce max-len on comments

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    // Are you sure | is not a typo for || ?
+    'no-bitwise': ['error'],
+
+    // Not enforcing props validation for React
+    'react/prop-types': 'off',
+
+    // Don't leave log statements littering the premises!
+    // 'no-console': ['error'],
+  },
+};
```
Comment (lumisphere902) [inline .eslintrc.js:125]: Trailing space (minor)

**Review by lumisphere902 — CHANGES_REQUESTED** [review]
Just a couple of small things, and then just need to fix the stuff that the script doesn't automatically do

File: .eslintrc.js (hunk 2 above)
Comment (reginawang99) [reply .eslintrc.js:125]: do you mean add an enter?

Comment (krashanoff) [reply .eslintrc.js:125]: Hey there, I'm not Timothy but that's right! Trailing space is a minor thing, but it helps with file versioning.

**Review by lumisphere902 — APPROVED** [review]
LGTM

**Pull request merged**